src/mc-fle2-rfds.c

```diff
@@ -0,0 +1,615 @@
+            goto fail;
+         }
+         spec->lower.set = true;
+         spec->lower.value = op.value;
+         spec->lower.included = op.op_type == FLE2RangeOperator_kGte;
+         break;
+      case FLE2RangeOperator_kLt:
+      case FLE2RangeOperator_kLte:
+         if (spec->upper.set) {
+            ERR_WITH_BSON (in, "unexpected duplicate bound %s", op.op_type_str);
+            goto fail;
+         }
+         spec->upper.set = true;
+         spec->upper.value = op.value;
+         spec->upper.included = op.op_type == FLE2RangeOperator_kLte;
+         break;
+      case FLE2RangeOperator_kNone:
+      default:
+         ERR_WITH_BSON (in, "unsupported operator type %s", op.op_type_str);
+         goto fail;
+         break;
+      }
+
+      if (spec->field) {
+         if (0 != strcmp (spec->field, op.field)) {
+            ERR_WITH_BSON (
+               in,
+               "unexpected field mismatch. Expected all fields to be "
+               "%s, but got %s",
+               spec->field,
+               op.field);
+            goto fail;
+         }
+      }
+
+      spec->nOps++;
+
+      if (spec->nOps == 1) {
+         spec->firstOp = op.op_type;
+      } else if (spec->nOps == 2) {
+         spec->secondOp = op.op_type;
+      } else {
+         ERR_WITH_BSON (
+            in, "expected 1 or 2 operators, got > 2: %s", op.op_type_str);
+         goto fail;
+      }
+
+      spec->field = op.field;
+   }
+
+   ok = true;
+fail:
+   return ok;
+}
+
+// mc_makeRangeFindPlaceholder creates a placeholder to be consumed by
+// libmongocrypt for encryption.
+bool
+mc_makeRangeFindPlaceholder (mc_makeRangeFindPlaceholder_args_t args,
+                             _mongocrypt_buffer_t *out,
+                             mongocrypt_status_t *status)
+{
+   BSON_ASSERT_PARAM (out);
+   BSON_ASSERT (status || true);
+
+   bool ok = false;
+   bson_t *edgesInfo = bson_new ();
+   bson_t *v = bson_new ();
+   bson_t *p = bson_new ();
+   _mongocrypt_buffer_init (out);
+
+#define TRY(stmt)                                          \
+   if (!(stmt)) {                                          \
+      CLIENT_ERR ("error appending BSON for placeholder"); \
+      goto fail;                                           \
+   }
+
+   // create edgesInfo.
+   if (!args.isStub) {
+      TRY (bson_append_iter (edgesInfo, "lowerBound", -1, &args.lowerBound));
+      TRY (BSON_APPEND_BOOL (edgesInfo, "lbIncluded", args.lbIncluded));
+      TRY (bson_append_iter (edgesInfo, "upperBound", -1, &args.upperBound));
+      TRY (BSON_APPEND_BOOL (edgesInfo, "ubIncluded", args.ubIncluded));
+      TRY (bson_append_iter (edgesInfo, "indexMin", -1, &args.indexMin));
+      TRY (bson_append_iter (edgesInfo, "indexMax", -1, &args.indexMax));
+      TRY (BSON_APPEND_DOCUMENT (v, "edgesInfo", edgesInfo));
+   }
+
+   // create v.
+   TRY (BSON_APPEND_INT32 (v, "payloadId", args.payloadId));
+   TRY (BSON_APPEND_INT32 (v, "firstOperator", args.firstOp));
```

Comment:
```suggestion
   TRY (BSON_APPEND_INT32 (v, "firstOperator", (int32_t) args.firstOp));
```
Address `warning: implicit conversion changes signedness: 'mc_FLE2RangeOperator_t' to 'int32_t' (aka 'int') [-Wsign-conversion]`.